src/txkube/test/test_authentication.py

```diff
@@ -95,19 +143,21 @@ def _authorized_request(self, token, headers):
 
         self.patch(
             os, "environ", {
-                b"KUBERNETES_SERVICE_HOST": b"example.invalid.",
+                b"KUBERNETES_SERVICE_HOST": kubernetesHost,
                 b"KUBERNETES_SERVICE_PORT": b"443",
             },
         )
 
         agent = authenticate_with_serviceaccount(
             reactor, path=serviceaccount.path,
         )
-        agent.request(b"GET", b"http://example.invalid.", headers)
-
+        d = agent.request(b"GET", b"http://" + kubernetesHost, headers)
+        if isinstance(d.result, Failure):
+            return d
```

Comment:
I think this should use the `testtools.twistedsupport.succeeded` (and `testtools.matchers.Always`) matcher:

    self.assertThat(d, succeeded(Always())

The `Always()` just matches any result after `succeeded` matches on a successful deferred.  This has the benefit of the nice reporting that you get from testtools matches and avoiding using `Deferred.result` directly (which looks public but it's really a bad idea to ever use it directly).

Reply:
I tried `self.assertThat(d, succeeded(Always()))` and got:

```
===============================================================================
[FAIL]
Traceback (most recent call last):
Failure: testtools.testresult.real._StringException: Traceback (most recent call last):
  File "/Users/crodrigues/txkube2/src/txkube/test/test_authentication.py", line 262, in test_other_headers_preserved
    request_bytes = self._authorized_request(token=token, headers=headers)
  File "/Users/crodrigues/txkube2/src/txkube/test/test_authentication.py", line 159, in _authorized_request
    self.assertThat(d, succeeded(Always()))
  File "/Users/crodrigues/venv-2.7/lib/python2.7/site-packages/testtools/testcase.py", line 498, in assertThat
    raise mismatch_error
testtools.matchers._impl.MismatchError: Success result expected on <Deferred at 0x1120f98c0 waiting on Deferred at 0x1120f9b00>, found no result instead

```

Reply:
Oh, huh.  I guess the point of the code is that _if_ the `Deferred` has failed, fail quickly.  But expected behavior is for it to have no result at all yet.

I wonder if `assertThat(d, Not(failed(Always())))` would work... I expect probably not.  Peeking at an in-progress `Deferred` is kind of a sneaky thing.  I'm going to sleep on this, maybe I'll remember some better trick.